### ISSUE TITLE
Authenticate through Nextcloud files app

### ISSUE THREAD

**Issue by lneugebauer** [lneugebauer/nextcloud-cookbook#4]
Use the [android library](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/android_library/index.html#nextcloud-android-library) to authenticate through the Nextcloud files app.

The prefered way of authenticating should be through the Nextcloud files app. As an alternative the user should still be able to authenticate with username and app password.

**Comment by lneugebauer** [comment]
Integrated nextcloud sso libraray in commit 09af350c9b5d1c7f43f3c81b5e9d91026ecdca5f. Currently sso is still non-functional. There seems to be a bug in the library (https://github.com/nextcloud/Android-SingleSignOn/issues/177) which prevents me from integrating it fully.

**Comment by p-fruck** [comment]
Hey there, I am currently getting started with nextcloud cookbook and searching for an companion app. I'm a little confused - The documentation and images say that it is possible to log in using the nextcloud files app. However, the button does not appear on my device (Nextcloud Files installed via F-Droid, this was causing errors previously). Also, I couldn't find any reference in the code. Is this already implemented and supposed to work on non-google devices?
Also, regarding the SSO API, there are other apps utilizing this API for SSO in Kotlin apps in case you are interested: https://codeberg.org/MicMun/nextcloud-cookbook/src/branch/master/app/src/main/java/de/micmun/android/nextcloudcookbook/nextcloudapi/Accounts.kt

**Comment by lneugebauer** [comment]
Hi @p-fruck, this sure is confusing! I originally started working on SSO integration but never got it to work. The screenshots are quite old and from that time.

Unfortunately, I haven't gotten around to working on it since then.

TL;DR is that you cannot currently use SSO for authentication. Once integrated, it will also work on non-Google devices.

**Comment by p-fruck** [comment]
@lneugebauer wow, that reply was quick :smile: Thanks for the explanation, I'll try to use the token login then ;)
